parceler-api/src/main/java/org/parceler/NonParcelRepository.java

```diff
@@ -265,6 +265,14 @@ public Parcelable buildParcelable(String input) {
         }
     }
 
+    static class ParcelableParcelableFactory implements Parcels.ParcelableFactory<Parcelable>{
+
+        @Override
+        public Parcelable buildParcelable(Parcelable input) {
+            return new ParcelableParcelable(input);
```

Comment:
This feels a little funny, but I guess it takes care of unwrapping the wrapped Parcelable.